tests/unit/tasks/models/jobsModel.spec.ts

```diff
@@ -0,0 +1,363 @@
+      setValue('ingestionTaskType', { tileMergeTask, tileSplitTask });
+
+      const rasterMapTestData = { ...testMetadata };
+      rasterMapTestData.productType = ProductType.RASTER_MAP;
+
+      getJobStatusMock.mockReturnValue({
+        id: jobId,
+        completed: true,
+        successful: true,
+        relativePath: `test/${ProductType.RASTER_MAP}`,
+        percentage: 80,
+        metadata: rasterMapTestData,
+        type: ingestionUpdateJobType,
+        status: OperationStatus.IN_PROGRESS,
+      });
+
+      getTaskMock.mockReturnValue({
+        id: taskId,
+        jobId: jobId,
+        type: tileMergeTask,
+        status: OperationStatus.FAILED,
+      });
+
+      // tasksManager = new TasksManager(
+      //   logger,
+      //   configMock,
+      //   syncClientMock,
+      //   jobManagerClientMock,
+      //   mapPublisherClientMock,
+      //   catalogClientMock,
+      //   linkBuilderMock,
+      //   metadataMergerMock
+      // );
+
+      // eslint-disable-next-line @typescript-eslint/ban-types
+      const tasksManagerWithHandlers = jobsManager as unknown as { handleUpdateIngestion: () => {}; handleNewIngestion: () => {} };
+      const handleUpdateIngestionSpy = jest.spyOn(tasksManagerWithHandlers, 'handleUpdateIngestion');
+      const handleNewIngestionSpy = jest.spyOn(tasksManagerWithHandlers, 'handleNewIngestion');
+      await jobsManager.completeJob(jobId, taskId);
+
+      expect(abortJobMock).toHaveBeenCalledTimes(1);
+      expect(updateJobStatusMock).toHaveBeenCalledTimes(1);
+      expect(updateJobStatusMock).toHaveBeenCalledWith(jobId, OperationStatus.FAILED, undefined, `Failed to update ingestion`);
+      expect(handleUpdateIngestionSpy).toHaveBeenCalledTimes(1);
+      expect(handleNewIngestionSpy).toHaveBeenCalledTimes(0);
+    });
+
+    it('should complete job once all tasks are successful for update-merge job-task', async function () {
+      setValue('ingestionUpdateJobType', ingestionUpdateJobType);
+      setValue('ingestionTaskType', { tileMergeTask, tileSplitTask });
+
+      const rasterMapTestData = { ...testMetadata };
+      rasterMapTestData.productType = ProductType.RASTER_MAP;
+
+      getJobStatusMock.mockReturnValue({
+        id: jobId,
+        isCompleted: true,
+        isSuccessful: true,
+        percentage: 90,
+        relativePath: `test/${ProductType.RASTER_MAP}`,
+        metadata: rasterMapTestData,
+        type: ingestionUpdateJobType,
+        successTasksCount: 3,
+        status: OperationStatus.IN_PROGRESS,
+      });
+
+      getTaskMock.mockReturnValue({
+        id: taskId,
+        jobId: jobId,
+        type: tileMergeTask,
+        status: OperationStatus.COMPLETED,
+      });
+
+      const catalogRecordId = 'a6fbf0dc-d82c-4c8d-ad28-b8f56c685a23';
+      findRecordMock.mockResolvedValue({
+        id: catalogRecordId,
+        metadata: {},
+      });
+
+      // tasksManager = new TasksManager(
```

Comment:
why the remarks?

Reply:
Pending

should have been deleted, i removed it, old test initialized task manager every test, i made it initialize once before all and its called job manager now instad of task manager , see old tests => https://github.com/MapColonies/discrete-overseer-creator/blob/master/tests/unit/tasks/models/tasksModel.spec.ts